### PR TITLE
Change data-dump js trigger from title text to id

### PR DESCRIPTION
**Pull request by sonalkr132**
Title text will be different for different locals
Closes: #1902

### CODE REVIEW

File: app/assets/javascripts/pages.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //data page
 
-$(document).ready(function() {
+if($("#data-dump").length) {
```
Comment (colszowka) [inline app/assets/javascripts/pages.js]: I'm not familiar with the codebase, but shouldn't this remain `document.ready` and maybe get this extra check as an additional condition within the ready callback function?

**Comment by colszowka** [comment]
@sonalkr132 Ha, makes sense :) I'm not sure about the swap of the document ready though, I think it should remain in place.

Also, does the site have some feature or js tests? Might be worthwile to have some kind of check that verifies the downloads actually work, but I can see how with the current code it might be too much setup effort of adding tests (i.e. hardcoded external call to S3, so in test env this would need to be wrapped somehow etc). Alternately, maybe some live production monitoring script might periodically visit the live site and try to download the dumps, just to make sure things don't break?

**Comment by sonalkr132** [comment]
> I'm not sure about the swap of the document ready though, I think it should remain in place.

Added it back.
> does the site have some feature or js tests?

Beside this, we also make external call for github stars and they do not have test either. We will have to include webmock/vcr as dependency for it. We will need word from @dwradcliffe to clarify whether he prefers a test on sensu or adding a dependency in app.

**Comment by colszowka** [comment]
@sonalkr132 Cool, thanks! webmock/vcr wouldn't help here since the http call is on the JS layer, you'd have to somehow wrap the actual call to behave differently in the test environment. Probably not worth the hassle if the existing test infrastructure does not make it easy to achieve that kind of thing, I'd think having a regular check against production that goes to the page in different locales and checks that the links are actually there would be a reasonable choice here

**Comment by sonalkr132** [comment]
> not worth the hassle if the existing test infrastructure does not make it easy to achieve that kind of thing

We can probably use toxiproxy (already a dependency) with [this](https://www.mccartie.com/tech/2016/03/14/modify-your-ruby-app's-dns-lookup.html). We will still need to add a driver for js.
BTW, you don't have to depend on data dump page for automation of downloads. You could just use the bucket url: https://s3-us-west-2.amazonaws.com/rubygems-dumps/?prefix=production/public_
We tend to use js only as last resort and like to think that a js feature not working will be non-critical issue.

I am merging this without tests.

**Pull request merged**